api/v1alpha1/tenantcontrolplane_types.go

```diff
@@ -98,11 +98,34 @@ type AddonSpec struct {
 	Enabled *bool `json:"enabled,omitempty"`
 }
 
+// KonnectivitySpec defines the spec for Konnectivity.
+type KonnectivitySpec struct {
+	// +kubebuilder:default=true
+	Enabled *bool `json:"enabled,omitempty"`
+	// Port of Konnectivity proxy server.
+	// +kubebuilder:default=8132
+	ProxyPort int32 `json:"proxyPort,omitempty"`
```

Comment:
I would try to keep the default value (known one) in order to facilitate the use of kamaji to new comers. But going to port "auto-assignation" could increase a lot the complexity and I do not think that it deserves the effort because that port should be exposed, therefore, it will depend on the infrastructure where kamaji is deployed

Reply:
The issue of a default value, since using a `NodePort`, would end up having multiple Tenant Control Plane w/ Konnectivity enabled, in which just the first one would work and be able to get a Service due to a collision.

I see two options, here:

1. nullable port, in order to get the allocated one by the Admin cluster
2. marking the field as required if the feature is enabled, with no default values

With the first one, we have to decide how the admin can retrieve that port: it could be stored in the `KonnectivityStatus`, as we're doing with the Kubeconfig.

With the latter one, unfortunately, we would need a webhook, and we don't have yet the support for it, and that would require an error handling in the controller.

Reply:
if the solution is not clear, let's open a new improvement task

Reply:
We have to address it right now since it's a new feature and we can shape the feature as we wish.

According to OAPIv3, there's no way on setting up a condition.

A possible workaround could be the following: allowing an empty object.

> ```yaml
> ...
> addons:
>   konnectivity:
>     host: "fqdn.io"
>     port: 30123
>   coreDNS: {}
>   kubeProxy: {}
> ```

It means that all of the addons are activated since there are options.

Instead:

> ```yaml
> ...
> addons:
>   konnectivity:
>     host: "fqdn.io"
>     port: 30123
>   coreDNS: {}
> ```

in this case `kubeProxy` is disabled because we don't have any option declared, it's `nil`.

The benefit of this approach is that we can make the port required only when the object is declared, and not on the condition of `enabled=true`.

This is a refactoring, that must be addressed before the `konnectivity` feature in a different commit of this PR.